### PR TITLE
Remove deprecated `ImageProvider` methods

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
Have to keep `loadBuffer` since `loadImage` isn't available on 3.7. 
Fixes https://github.com/flutter/flutter/issues/105336

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [ ] I signed the [CLA].
- [ ] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by LongCatIsLooong** [comment]
Updating the implementation now that https://github.com/flutter/packages/pull/4731 has landed.

**Comment by LongCatIsLooong** [comment]
@stuartmorgan is a version bump needed for this change?

File: packages/flutter_image/test/network_test.dart [hunk 1]
```diff
@@ -160,12 +160,12 @@ void assertThatImageLoadingFails(
 void assertThatImageLoadingSucceeds(
   NetworkImageWithRetry subject,
 ) {
-  final ImageStreamCompleter completer = subject.load(
+  final ImageStreamCompleter completer = subject.loadBuffer(
     subject,
     // TODO(cyanglaz): migrate to use the new APIs
     // https://github.com/flutter/flutter/issues/105336
     // ignore: deprecated_member_use
```
Comment (stuartmorgan) [inline packages/flutter_image/test/network_test.dart:167]: Same question here.

File: packages/flutter_image/lib/network.dart [hunk 2]
```diff
@@ -95,10 +99,13 @@ class NetworkImageWithRetry extends ImageProvider<NetworkImageWithRetry> {
...
 
   @override
-  // TODO(cyanglaz): migrate to use the new APIs
-  // https://github.com/flutter/flutter/issues/105336
-  // ignore: deprecated_member_use
-  ImageStreamCompleter load(NetworkImageWithRetry key, DecoderCallback decode) {
+  ImageStreamCompleter loadBuffer(
+    NetworkImageWithRetry key,
+    // TODO(cyanglaz): migrate to use the new APIs
+    // https://github.com/flutter/flutter/issues/105336
```
Comment (stuartmorgan) [inline packages/flutter_image/lib/network.dart:105]: This issue is about APIs deprecated a year ago, so shouldn't still be applicable to new code. If there's a new migration needed, there should be a new issue describing it clearly (the old API, the new API, and the version it's in).

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Yes, this changes production code so it needs a version bump. See https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version for the list of cases that don't.

File: packages/flutter_image/lib/network.dart (hunk 2 above)
Comment (LongCatIsLooong) [reply packages/flutter_image/lib/network.dart]: Created https://github.com/flutter/flutter/issues/132856

File: packages/flutter_image/test/network_test.dart [hunk 3]
```diff
@@ -140,12 +140,12 @@ void assertThatImageLoadingFails(
   NetworkImageWithRetry subject,
   List<FlutterErrorDetails> errorLog,
 ) {
-  final ImageStreamCompleter completer = subject.load(
+  final ImageStreamCompleter completer = subject.loadBuffer(
     subject,
     // TODO(cyanglaz): migrate to use the new APIs
     // https://github.com/flutter/flutter/issues/105336
     // ignore: deprecated_member_use
```
[the remark this answers is not in the document]
Comment (LongCatIsLooong) [reply packages/flutter_image/test/network_test.dart:147]: Yeah the up-to-date APIs are only available in >v3.7.0 unfortunately. Included in the new issue.

**Review by stuartmorgan — APPROVED** [review]
LGTM, thanks!

Alternately, it's also totally fine to just make the minimum version 3.10 and fully eliminate the deprecated APIs (per https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#supported-flutter-versions)

**Pull request merged**